sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/SymbolicExecution/ExplodedGraphTests.cs

```diff
@@ -925,26 +959,35 @@ private ExplodedGraphContext(MethodDeclarationSyntax mainMethod, SemanticModel s
                 this.ExplodedGraph.ExitBlockReached += (sender, args) => { this.NumberOfExitBlockReached++; };
             }
 
-            public ISymbol GetSymbol(string identifier)
+            public enum SymbolType
             {
-                var varDeclarator = this.MainMethod
-                    .DescendantNodes()
-                    .OfType<VariableDeclaratorSyntax>()
-                    .FirstOrDefault(d => d.Identifier.ToString() == identifier);
+                Variable,
+                Identifier,
+                Declaration
+            }
 
-                if (varDeclarator != null)
+            public ISymbol GetSymbol(string identifier, SymbolType st = SymbolType.Variable)
+            {
+                var expression = st switch
                 {
-                    return this.SemanticModel.GetDeclaredSymbol(varDeclarator);
-                }
-
-                var declarationExpression = this.MainMethod
-                    .DescendantNodes()
-                    .OfType<DeclarationExpressionSyntax>()
-                    .Where(d => d.Designation is SingleVariableDesignationSyntax)
-                    .Select(d => (SingleVariableDesignationSyntax)d.Designation)
-                    .First(d => d.Identifier.Text == identifier);
+                    SymbolType.Variable => this.MainMethod
+                        .DescendantNodes()
+                        .OfType<VariableDeclaratorSyntax>()
+                        .First(d => d.Identifier.ToString() == identifier),
+                    SymbolType.Identifier => (CSharpSyntaxNode) this.MainMethod
```

Comment:
`CSharpSyntaxNode` is needed to help the compiler figure out the type of the switch expression